### ISSUE TITLE
We need to generate material share events 

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/resources-portal#262]
### Context

Material share events are our plan for tracking what happens to a material request and when.

### Problem or idea

We have a material share model, but that's as far as we've gotten with it in the backend. I don't think we've discussed it much since our initial database designs. My understanding is that as the material request gets updated in various ways by various users, we will automatically generate material share events.

@dvenprasad shared this image of a request flow in slack:

EDIT: Added an edge from `Awaiting MTA from Sharer` to `Cancelled`
<img width="1190" alt="Screen Shot 2020-08-03 at 11 10 19 AM" src="https://user-images.githubusercontent.com/15315514/89197749-316a5500-d57a-11ea-8aa7-2f602e9778c8.png">

It seems to me that basically every edge of that graph should generate a material share event. In addition, we should also generate a material share event when the request is assigned to another user. Am I missing any? Is this too many?

### Solution or next step

I think we should hash out exactly what needs to trigger material share events first @davidsmejia @dvenprasad @Philinphiladelphia. We want to make sure that we track anything that might help us
* Give credit to sharers
* Figure out that someone was negligent or otherwise not a good sharer
* Be useful in generating a statistic in a report for either the grants team or Deepa for usability evaluation purposes.
* Figure out where in a pipeline requests get stuck so we can figure out ways to keep them moving smoothly.

Once we're done brainstorming/hashing that out, this issue can then become about implementation.

**Comment by kurtwheeler** [comment]
I'll also note that the current model doesn't have a link to a material request but probably should.

**Comment by kurtwheeler** [comment]
These are the events I have come up with so far:
```
REQUEST_CANCELLED
REQUEST_REJECTED
REQUEST_ACCEPTED
REQUESTER_IRB_ADDED
REQUESER_MTA_ADDED
REQUESTER_PAYMENT_METHOD_ADDED
REQUESTER_PAYMENT_NOTES_ADDED
SHARER_MTA_ADDED
REQUEST_IN_FULFILLMENT
REQUEST_FULFILLED
REQUEST_VERIFIED_FULFILLED
REQUEST_FULFILLMENT_NOTE_ADDED
REQUEST_REASSIGNED
REQUEST_ISSUE_OPENED
REQUEST_ISSUE_MODIFIED
REQUEST_ISSUE_CLOSED
MATERIAL_MTA_REQUIREMENTS_CHANGED
MATERIAL_IRB_REQUIREMENTS_CHANGED
MATERIAL_SHIPPING_REQUIREMENTS_CHANGED
```

Am I missing any? (I don't think I need one for a request being created. That's contained in the request.created_at, although I could make one just so that if we ever listed them out in order a full story is shown without special handling for created.)

I've left off events related to grants and teams because I don't think we need to track how those evolve over time.

**Comment by kurtwheeler** [comment]
Closed by #457
